rust/frontend/src/impls/service_based_frontend.rs

```diff
@@ -377,6 +378,22 @@ impl ServiceBasedFrontend {
             .map(|count| count as u32)
     }
 
+    /// Parse a collection name to see if it matches the pattern "<tenant_resource_name>:<database_name>:<collection_name>"
+    /// Returns Some((tenant_resource_name, database_name, collection_name)) if the pattern matches, None otherwise
+    fn parse_crn(&self, crn: &str) -> Option<(String, String, String)> {
```

Comment:
I think this parse method should be part of object creation and not in this file. so CollectionResourceName::new() calls this internally, and we can reduce the duplication with the validator in types/